File: chapters/euclidean_algorithm/code/cs/EuclideanAlgorithm.cs
// submitted by Julian Schacher (jspp)
using System;

namespace EuclideanAlgorithm
{
    public class EuclideanAlgorithm
    {
        public int EuclidSub(int a, int b)
        {
            // Math.Abs for negative number support
            a = Math.Abs(a);
            b = Math.Abs(b);

            while (a != b)
            {
                if (a > b)
                    a = a - b;
                else
                    b = b - a;
            }

            return a;
        }

        public int EuclidMod(int a, int b)
        {
            // Math.Abs for negative number support
            a = Math.Abs(a);
            b = Math.Abs(b);

            while (b != 0)
            {
                var temp = b;
                b = a % b;
                a = temp;
            }

            return a;
        }
    }
}

File: chapters/euclidean_algorithm/code/swift/euclidean_algorithm.swift
func euclidSub(a: Int, b: Int) -> Int {
    var A = abs(a)
    var B = abs(b)
    
    while (A != B) {
        if (A > B) {
            A -= B
        } else {
            B -= A
        }
    }
    
    return A
}


func euclidMod(a: Int, b: Int) -> Int {
    var A = abs(a);
    var B = abs(b);
    
    while (B != 0) {
        let temp = B
        B = A % B
        A = temp
    }
    
    return A
}




func main() {
    print(euclidMod(a: 64 * 67, b: 64 * 81))
    print(euclidSub(a: 128 * 12, b: 128 * 77))
}


main()
